File: src/main/webapp/app/shared/model/DateRangeDTO.ts
import dayjs from 'dayjs/esm';

export interface IDateRangeDTO {
  startDate?: dayjs.Dayjs | null;
  endDate?: dayjs.Dayjs | null;
}

export class DateRangeDTO implements IDateRangeDTO {
  constructor(public startDate?: dayjs.Dayjs, public endDate?: dayjs.Dayjs) {}
}

File: src/main/java/com/bits/hr/service/dto/DateRangeDTO.java
package com.bits.hr.service.dto;

import java.time.LocalDate;

public class DateRangeDTO {

    private LocalDate startDate;
    private LocalDate endDate;

    public DateRangeDTO() {}

    public LocalDate getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDate startDate) {
        this.startDate = startDate;
    }

    public LocalDate getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDate endDate) {
        this.endDate = endDate;
    }

    @Override
    public String toString() {
        return "DateRangeDTO{" + "startDate=" + startDate + ", endDate=" + endDate + '}';
    }
}
